lib/api-helper/machine/handlerRegistrations.ts

```diff
@@ -713,8 +713,42 @@ function toEditModeOrFactory<P>(ctr: CodeTransformRegistration<P>,
         description);
 }
 
-function gitBranchCompatible(name: string): string {
-    return name.replace(/\s+/g, "_"); // What else??
+/**
+ * Takes a potential git branch name and returns a legalised iteration of it
+ *
+ * @param name the git branch name to sanitise.
+ */
+export function gitBranchCompatible(name: string): string {
+    // handles spaces and .. ~ : ^ ? * [ @{
+    let branchName = name.replace(/\s+|(\.\.)+|~+|:+|\^+|\?+|\*+|\[+|(\@\{)+/g, "_");
+
+    // handles double slashes
+    branchName = branchName.replace(/(\/\/)+|(\\)+/g, "/");
+
+    // handles back slashes
+    branchName = branchName.replace(/\\+/g, "/");
```

Comment:
Wouldn't the previous replace take care of this? In other words, I think this replacement can be removed.